### PR TITLE
[SM-1292] Add secret view dialog

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/SM-1292

## 📔 Objective

The objective of this PR is to add a secret view dialog for a read-only view of a secret.

### Before
image

### After
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-view-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<form [formGroup]="formGroup">
+  <bit-dialog
+    dialogSize="large"
+    [loading]="loading"
+    [title]="'viewSecret' | i18n | uppercase"
```
Comment (coltonhurst) [inline bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-view-dialog.component.html:5]: Any reason we are making this uppercase? I believe Create and Edit for secrets (+ edit for projects) aren't this way, unless that's going to be changed?

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.ts [hunk 2]
```diff
@@ -115,6 +116,14 @@ export class SecretsListComponent implements OnDestroy {
     return aProjects[0]?.name.localeCompare(bProjects[0].name);
   };
 
+  protected editSecret(secret: SecretListView) {
```
Comment (coltonhurst) [inline bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.ts:119]: I was trying to think if there was a better name for this to convey the full scope of what could happen, but I can't think of anything at the moment. I thought about `clickSecret` but that's not descriptive either. 🤦‍♂️ So all good, just was mentioning I thought about it.

**Review by coltonhurst — CHANGES_REQUESTED** [review]
One real comment, other than that it looks great. Thanks!

File: bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-view-dialog.component.html (hunk 1 above)
Comment (Thomas-Avery) [reply bitwarden_license/bit-web/src/app/secrets-manager/secrets/dialog/secret-view-dialog.component.html:5]: It is what the design calls for. Currently, all of those mentioned are upper case in our design documents. 

Would you like me to update the create and edit in this PR?

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.ts (hunk 2 above)
Comment (Thomas-Avery) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.ts:119]: At one point I had `handleSecretInteraction` but it seemed too wordy, so I went with this.

Open to suggestions. Since this isn't a public method, we only have to convey things internal to this component's context.

Comment (coltonhurst) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.ts:119]: No worries, feel free to leave as is, thank you 👍

**Review by coltonhurst — DISMISSED** [review]

**Review by coltonhurst — APPROVED** [review]
